### PR TITLE
Stop installation of files in user directory [REVPI-3094]

### PR DESCRIPTION
**Pull request by hjokunbus**
Project files must not be installed in /home directory. Instead documentation and binaries have been placed into either
`/usr/share/doc` or `/usr/lib`. The resulting installation paths look like this now

```
$ sudo make install
[100%] Built target patch_eeprom
[100%] Built target lan9514.bin
Install the project...
-- Install configuration: ""
-- Installing: /usr/local/etc/avahi/services/revpi-ssh.service
-- Installing: /usr/lib/revpi-tools/patch_eeprom
-- Installing: /usr/share/doc/revpi-tools/README
-- Installing: /usr/share/doc/revpi-tools/patch_eeprom.c
-- Installing: /usr/share/doc/revpi-tools/enable_relay_watchdog.py
-- Installing: /usr/share/doc/revpi-tools/disable_relay_watchdog.py
-- Installing: /usr/share/doc/revpi-tools/revpi-connect-watchdog.sh
-- Installing: /lib/systemd/system/firstboot.service
-- Installing: /usr/local/share/revpi/firstboot/resize-fs.sh
-- Installing: /usr/local/sbin/revpi_buster_upgrade_fix_nodered.sh
-- Installing: /usr/local/sbin/revpi_buster_fix_pictory.sh
-- Installing: /usr/local/sbin/revpi_buster_fixup_pkgs.py
-- Installing: /usr/local/sbin/revpi-config
-- Installing: /usr/local/share/man/man1/revpi-config.1
-- Installing: /usr/local/share/revpi/revpi-functions
-- Installing: /usr/local/sbin/revpi-factory-reset
-- Installing: /usr/local/etc/profile.d/revpi-factory-reset.sh
-- Installing: /usr/local/share/man/man8/revpi-factory-reset.8
-- Installing: /usr/local/bin/revpi-sos
-- Installing: /usr/local/share/sosreport/sos/plugins/revpi.py
-- Installing: /usr/local/share/man/man1/revpi-sos.1
-- Installing: /lib/systemd/system/pileft-quirks.service
-- Installing: /lib/systemd/system/piright-quirks.service
-- Installing: /lib/systemd/system/hdmi-disable.service
-- Installing: /usr/local/sbin/pibridge-shutdown
-- Installing: /usr/local/share/man/man8/pibridge-shutdown.8
-- Installing: /usr/local/sbin/ks8851-set-mac
-- Installing: /usr/local/sbin/lan95xx-set-mac
-- Installing: /usr/local/share/man/man8/ks8851-set-mac.8
-- Installing: /usr/local/share/revpi/lan9514.bin
-- Installing: /lib/udev/rules.d/50-revpi.rules
```

### CODE REVIEW

File: examples-watchdog/CMakeLists.txt [hunk 1]
```diff
@@ -3,9 +3,9 @@ add_executable(patch_eeprom patch_eeprom.c)
 target_link_libraries(patch_eeprom ftdi1)
 
 install(TARGETS patch_eeprom
-  DESTINATION /home/pi/connect/
+  DESTINATION /usr/lib/revpi-tools/
```
Comment (nbuchwitz) [inline examples-watchdog/CMakeLists.txt:6]: Suggestion: Use `CMAKE_INSTALL_*` variables:

```suggestion
  DESTINATION ${CMAKE_INSTALL_LIBDIR}/revpi-tools
```

https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html#result-variables

File: examples-watchdog/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,9 @@ add_executable(patch_eeprom patch_eeprom.c)
 target_link_libraries(patch_eeprom ftdi1)
```
Comment (nbuchwitz) [inline examples-watchdog/CMakeLists.txt:4]: Suggestion: Use `CMAKE_INSTALL_*` variables. For this we need to include `GNUInstallDirs`

```suggestion

include(GNUInstallDirs)
```

File: examples-watchdog/CMakeLists.txt [hunk 3]
```diff
@@ -3,9 +3,9 @@ add_executable(patch_eeprom patch_eeprom.c)
...
 
 install(TARGETS patch_eeprom
-  DESTINATION /home/pi/connect/
+  DESTINATION /usr/lib/revpi-tools/
 )
 
 install(FILES README patch_eeprom.c enable_relay_watchdog.py disable_relay_watchdog.py revpi-connect-watchdog.sh
-  DESTINATION /home/pi/connect/
+  DESTINATION /usr/share/doc/revpi-tools/
```
[the remark this answers is not in the document]
Comment (hjokunbus) [reply examples-watchdog/CMakeLists.txt]: Done

**Comment by hjokunbus** [comment]
Suggested changes have been applied. Please note the path of Avahi services unrelated to this pull request.

```
$ sudo cmake --install . --prefix /usr
-- Install configuration: ""
-- Installing: /usr/etc/avahi/services/revpi-ssh.service
-- Installing: /usr/lib/revpi-tools/patch_eeprom
-- Installing: /usr/share/doc/revpi-tools/README
-- Installing: /usr/share/doc/revpi-tools/patch_eeprom.c
-- Installing: /usr/share/doc/revpi-tools/enable_relay_watchdog.py
-- Installing: /usr/share/doc/revpi-tools/disable_relay_watchdog.py
-- Installing: /usr/share/doc/revpi-tools/revpi-connect-watchdog.sh
-- Up-to-date: /lib/systemd/system/firstboot.service
-- Installing: /usr/share/revpi/firstboot/resize-fs.sh
-- Installing: /usr/sbin/revpi_buster_upgrade_fix_nodered.sh
-- Installing: /usr/sbin/revpi_buster_fix_pictory.sh
-- Installing: /usr/sbin/revpi_buster_fixup_pkgs.py
-- Installing: /usr/sbin/revpi-config
-- Installing: /usr/share/man/man1/revpi-config.1
-- Installing: /usr/share/revpi/revpi-functions
-- Installing: /usr/sbin/revpi-factory-reset
-- Installing: /usr/etc/profile.d/revpi-factory-reset.sh
-- Installing: /usr/share/man/man8/revpi-factory-reset.8
-- Installing: /usr/bin/revpi-sos
-- Installing: /usr/share/sosreport/sos/plugins/revpi.py
-- Installing: /usr/share/man/man1/revpi-sos.1
-- Up-to-date: /lib/systemd/system/pileft-quirks.service
-- Up-to-date: /lib/systemd/system/piright-quirks.service
-- Up-to-date: /lib/systemd/system/hdmi-disable.service
-- Installing: /usr/sbin/pibridge-shutdown
-- Installing: /usr/share/man/man8/pibridge-shutdown.8
-- Installing: /usr/sbin/ks8851-set-mac
-- Installing: /usr/sbin/lan95xx-set-mac
-- Installing: /usr/share/man/man8/ks8851-set-mac.8
-- Installing: /usr/share/revpi/lan9514.bin
-- Up-to-date: /lib/udev/rules.d/50-revpi.rules
```

**Review by nbuchwitz — APPROVED** [review]

**Review by RamiGspo — APPROVED** [review]

**Comment by nbuchwitz** [comment]
I'll have a meeting with them on Thursday and will mention this change

**Pull request merged**